.github/workflows/CI.yml

```diff
@@ -29,163 +29,112 @@ jobs:
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v1
-
-      - name: Setup Node
-        uses: actions/setup-node@v1
-        with:
-          node-version: 18.x
-
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - uses: actions/cache@v2
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
-        with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-
-      - name: Install dependencies
-        run: yarn --ignore-engines
-
-      - name: Typecheck
-        run: yarn check-types
-
-      - name: Run unit tests
-        run: yarn unit-test
-
-      # TODO: fix integration tests -- they currently fail
-      # - name: Run integration tests
-      #  run: yarn integration-test
-
-      # TODO: reintegrate code coverage once we get nyc working with hardhat
-
-      # TODO: fix linter and then reintegrate using current ts version
-      # - name: Run linter
-      #   run: yarn lint
-
-  code-quality-checks:
-    name: Code Quality Checks
-    runs-on: ubuntu-latest
-
-    steps:
-      - name: Checkout master
         uses: actions/checkout@v3
-        with:
-          ref: master
-          fetch-depth: 0
-
-      - name: Checkout branch
-        run: git checkout ${{ github.ref_name }}
 
       - name: Setup Node
         uses: actions/setup-node@v3
         with:
-          node-version: 18.x
-
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
+          node-version: ${{ matrix.node }}
 
-      - uses: actions/cache@v3
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      - name: Cache dependencies
+        id: cache
+        uses: actions/cache@v3
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          path: '**/node_modules'
+          key: ${{ matrix.node }}-${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
 
       - name: Install dependencies
         run: yarn --ignore-engines
 
+      - name: Init Db
+        run: cd packages/commonwealth && npx sequelize db:migrate
+
       - name: Run linter
         run: yarn lint-branch
 
-      - name: Run formatter
-        run: yarn format-check
+      - name: Typecheck
```

Comment:
Can we run the linter after the typecheck + tests? Just so that linter errors don't prevent the tests from running.